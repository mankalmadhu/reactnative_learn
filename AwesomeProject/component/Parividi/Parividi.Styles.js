import {
 StyleSheet
} from 'react-native';


const parividiVinyasa = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
});

export default parividiVinyasa;