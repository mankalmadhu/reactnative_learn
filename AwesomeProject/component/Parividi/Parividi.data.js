/////////////////////////////////////
const prakaranaMahiti = [
  {
    "gender": "male",
    "name": {
      "title": "mr",
      "first": "aiden",
      "last": "lucas"
    },
  
    "picture": {
      "large": "https://randomuser.me/api/portraits/men/4.jpg",
      "medium": "https://randomuser.me/api/portraits/med/men/4.jpg",
      "thumbnail": "https://randomuser.me/api/portraits/thumb/men/4.jpg"
    }
  },
  {
    "gender": "male",
    "name": {
      "title": "mr",
      "first": "mario",
      "last": "walters"
    },
   
    "picture": {
      "large": "https://randomuser.me/api/portraits/men/47.jpg",
      "medium": "https://randomuser.me/api/portraits/med/men/47.jpg",
      "thumbnail": "https://randomuser.me/api/portraits/thumb/men/47.jpg"
    }
  },
  {
    "gender": "male",
    "name": {
      "title": "mr",
      "first": "joseph",
      "last": "lambert"
    },
   
    "picture": {
      "large": "https://randomuser.me/api/portraits/men/13.jpg",
      "medium": "https://randomuser.me/api/portraits/med/men/13.jpg",
      "thumbnail": "https://randomuser.me/api/portraits/thumb/men/13.jpg"
    }
}]
////////////////////////////////////////
export default prakaranaMahiti;