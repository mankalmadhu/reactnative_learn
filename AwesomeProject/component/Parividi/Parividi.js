import {ListView} from 'react-native';
import React from 'react';

import Prakarana from '../Prakarana/Prakarana';
import parividiVinyasa from './Parividi.Styles';
import prakaranaMahiti from './Parividi.data';

class Parividi extends React.Component {
  constructor(props) {
    super(props);

    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this.state = {
      dataSource: ds.cloneWithRows(prakaranaMahiti),
    };
  }
  render() {
    return (
      <ListView
        style={parividiVinyasa.container}
        dataSource={this.state.dataSource}
        renderRow={(prakaranaMahiti) => <Prakarana {...prakaranaMahiti} />}
      />
    );
  }
}

export default Parividi;