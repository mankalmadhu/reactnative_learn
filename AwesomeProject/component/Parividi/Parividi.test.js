import 'react-native';
import React from 'react';
import Parividi from './Parividi';

import Renderer from 'react-test-renderer';

jest.mock('../Prakarana/Prakarana');

test("Parividi Render" , () => {

   const parividiRenderer = Renderer.create(
       <Parividi/>
   ).toJSON();

   expect(parividiRenderer).toMatchSnapshot();

});