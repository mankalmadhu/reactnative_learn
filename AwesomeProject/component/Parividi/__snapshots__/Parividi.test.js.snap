// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parividi Render 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <View>
    <div>
      Prakarana 
      aiden lucas
    </div>
    <div>
      Prakarana 
      mario walters
    </div>
    <div>
      Prakarana 
      joseph lambert
    </div>
  </View>
</RCTScrollView>
`;
